sql/hive/src/test/scala/org/apache/spark/sql/hive/MetastoreRelationSuite.scala

```diff
@@ -36,4 +38,16 @@ class MetastoreRelationSuite extends SparkFunSuite {
     // No exception should be thrown
     relation.toJSON
   }
+
+  test("SPARK-17409: Do Not Optimize Query in CTAS More Than Once") {
+    withTable("bar") {
+      withTempView("foo") {
+        sql("select 0 as id").createOrReplaceTempView("foo")
+        sql("CREATE TABLE bar AS SELECT * FROM foo group by id")
```

Comment:
Also put the comment of https://github.com/apache/spark/pull/15459/files#diff-5d2ebf4e9ca5a990136b276859769289R1626 at here? Also mention that this test is for hive format table?